packages/ilmomasiina-backend/src/routes/utils.ts

```diff
@@ -0,0 +1,34 @@
+import { StringifyApi } from '@tietokilta/ilmomasiina-models/src/schema/utils';
+
+export function toDate<T extends string | null | undefined>(s: string | T):
+T extends string ? Date : Exclude<T, string> {
+  // @ts-ignore
+  return (typeof s === 'string') ? new Date(s) : s;
+}
+
+export function toDateIfDefined(s?: string | null): Date | null | undefined {
+  return typeof s === 'string' ? new Date(s) : s;
+}
+
+export function nullsToUndef<T>(val: T | null): T | undefined {
+  return val !== null ? val : undefined;
+}
+
+export function stringifyDates<T extends object>(o: T): StringifyApi<T> {
+  const s = {};
+  for (const [key, val] of Object.entries(o)) {
+    if (val instanceof Date) {
+      // @ts-ignore
+      s[key] = val.toISOString();
+    } else if (typeof val !== 'object' || val == null) {
+      // @ts-ignore
+      s[key] = val;
+    } else {
+      // @ts-ignore
+      s[key] = stringifyDates(val);
+    }
+  }
+
+  // @ts-ignore
+  return o;
+}
```

Comment:
- Bug: Return `s` instead of `o`
- Use descriptive variable names (very much because of the above)
- Add a doc comment
- Avoid `@ts-ignore`; use `any` when necessary

```suggestion
/** Recursively converts all `Date` objects in the given object to strings. */
export function stringifyDates<T extends object>(obj: T): StringifyApi<T> {
  const result: any = {};
  for (const [key, val] of Object.entries(obj)) {
    if (val instanceof Date) {
      result[key] = val.toISOString();
    } else if (Array.isArray(val)) {
      result[key] = val.map(stringifyDates);
    } else if (typeof val === 'object' && val !== null) {
      result[key] = stringifyDates(val);
    } else {
      result[key] = val;
    }
  }
  return result;
}
```

Reply:
Resolved in 8d88854